src/routing/mod.rs

```diff
@@ -466,10 +466,29 @@ async fn handle_message(dispatcher: Arc<Dispatcher>, bytes: Bytes, sender: Socke
     };
     let _span_guard = span.enter();
 
-    let message_type = match WireMsg::deserialize(bytes) {
+    let wire_msg = match WireMsg::from(bytes) {
```

Comment:
Are we deserialising the whole message here as the point is not to and just read dst an dif it's us then process it (deserialise) or forward it. 
Also msg_id to be read from msg without deserializing the whole message and added to filter / checked we have handled it.

Reply:
`WireMsg::from` only restore the header part, which contains the msg_id
`WireMsg::deserialize` or `WireMsg::to_message` will deserialize the whole message